plugins/polio/js/src/domains/Campaigns/MainDialog/CreateEditDialog.tsx

```diff
@@ -68,8 +71,11 @@ const CreateEditDialog: FunctionComponent<Props> = ({
     const handleSubmit = useCallback(
         (values, helpers) => {
             saveCampaign(convertEmptyStringToNull(values), {
-                onSuccess: () => {
+                onSuccess: result => {
                     setIsUpdated(true);
+                    if (!selectedCampaignId) {
+                        setSelectedCampaignId(result.id);
+                    }
```

Comment:
you could use react-query to save the result directly in the cache. It would save a `useState` and maybe an API call, but this works fine